framework/src/consumption/lib/redshift-serverless/redshift-serverless-namespace.ts

```diff
@@ -0,0 +1,185 @@
+import { randomBytes } from 'crypto';
+import { Names, RemovalPolicy, Stack, Tags } from 'aws-cdk-lib';
+import { Effect, IRole, PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import { Key } from 'aws-cdk-lib/aws-kms';
+import { ISecret, Secret } from 'aws-cdk-lib/aws-secretsmanager';
+import { AwsCustomResource, AwsCustomResourcePolicy, PhysicalResourceId } from 'aws-cdk-lib/custom-resources';
+import { Construct } from 'constructs';
+import { RedshiftServerlessNamespaceProps } from './redshift-serverless-namespace-props';
+import { Context, TrackedConstruct, TrackedConstructProps } from '../../../utils';
+
+/**
+ * Create a Redshift Serverless Namespace with the admin credentials stored in Secrets Manager
+ * @see https://awslabs.github.io/data-solutions-framework-on-aws/docs/constructs/library/Consumption/redshift-serverless
+ *
+ * @example
+ * const namespace = new dsf.consumption.RedshiftServerlessNamespace(this, 'DefaultServerlessNamespace', {
+ *    dbName: 'defaultdb',
+ *    name: 'default'
+ * });
+ */
+export class RedshiftServerlessNamespace extends TrackedConstruct {
+  /**
+   * Created namespace
+   */
+  readonly cfnResource: AwsCustomResource;
+
+  /**
+   * The name of the created namespace
+   */
+  readonly namespaceName: string;
+
+  /**
+   * The created Secrets Manager secret containing the admin credentials
+   */
+  readonly adminSecret: ISecret;
+
+  /**
+   * The roles attached to the namespace
+   */
+  readonly roles: Record<string, IRole>;
+
+  /**
+   * KMS key used by the namespace to encrypt its data
+   */
+  readonly namespaceKey: Key;
+
+  /**
+   * The name of the database
+   */
+  readonly dbName: string;
+
+  /**
+   * The ARN of the created namespace
+   */
+  readonly namespaceArn: string;
+
+  /**
+   * The ID of the created namespace
+   */
+  readonly namespaceId: string;
+
+  private readonly removalPolicy: RemovalPolicy;
+  private readonly currentStack: Stack;
+  private namespaceParameters: { [key:string]: any };
+
+  constructor(scope: Construct, id: string, props: RedshiftServerlessNamespaceProps) {
+    const trackedConstructProps: TrackedConstructProps = {
+      trackingTag: RedshiftServerlessNamespace.name,
+    };
+
+    super(scope, id, trackedConstructProps);
+
+    this.roles = {};
+    this.currentStack = Stack.of(this);
+
+    if (props.iamRoles) {
+      for (const role of props.iamRoles) {
+        this.roles[role.roleArn] = role;
+      }
+    }
+
+    if (props.defaultIAMRole) {
+      this.roles[props.defaultIAMRole.roleArn] = props.defaultIAMRole;
+    }
+
+    this.dbName = props.dbName;
+    this.removalPolicy = Context.revertRemovalPolicy(scope, props.removalPolicy);
+    const logExports: string[] = props.logExports || [];
+    this.namespaceName = `${props.name}-${Names.uniqueResourceName(this, { maxLength: 64 - (props.name.length + 1) }).toLowerCase()}`;
+
+    this.namespaceKey = props.kmsKey || new Key(this, 'DefaultNamespaceKey', { enableKeyRotation: true });
```

Comment:
can you change this to `??` its more robust, we want to avoid the `falsy` and adopt a `nullish` behavior, because that's what we are testing for.